File: TikraqApp/src/app/model/Calificacion.ts
export class Calificacion {
  id: number;
  rating: number;
  usuariosUsuarioid: { id: number };

  constructor(id: number =0, rating: number, usuariosUsuarioid: { id: number }) {
    this.id = id;
    this.rating = rating;
    this.usuariosUsuarioid = usuariosUsuarioid;
  }
}

File: src/main/java/com/example/tikraq/entities/Calificacion.java
package com.example.tikraq.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "calificaciones")
public class Calificacion {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "calificacionId", nullable = false)
    private Integer id;

    private Integer rating;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "usuarios_usuarioid")
    private Usuario usuariosUsuarioid;
}
